### ISSUE TITLE
Reader ignores first newline in value

### ISSUE THREAD

**Issue by paslandau** [thephpleague/csv#80]
I came across a weird behaviour while evaluation this package. When reading a CSV file or string via `Reader`, the first newline in a value of each line will be ignored.

Code to reproduce:

````php
$input = <<<EOS
"line 1 field 1 with crlf: > \r\n < second crlf: > \r\n <, line 1 field 2 with crlf: > \r\n < second crlf: > \r\n <"
"line 2 field 1 with crlf: > \r\n < second crlf: > \r\n <, line 2 field 2 with crlf: > \r\n < second crlf: > \r\n <"
EOS;

$reader = Reader::createFromString($input);
$reader->setNewline("\r\n");
foreach($reader as $r){
    var_dump($r);
}
````

**Output**
````
array(1) {
  [0] =>
  string(104) "line 1 field 1 with crlf: >  < second crlf: > 
 <, line 1 field 2 with crlf: > 
 < second crlf: > 
 <"
}
array(1) {
  [0] =>
  string(104) "line 2 field 1 with crlf: >  < second crlf: > 
 <, line 2 field 2 with crlf: > 
 < second crlf: > 
 <"
}
````

Please note the first ">  <" (without \r\n) in the beginning of each csv line in the output.

The problem does only occur when `SplFileObject::DROP_NEW_LINE` ist defined as flag -- unfortunately I could not find away to "unset" this flag from outside of the class since it is automatically added in the `Controls` class when setting the flags:

````php
    /**
     * Set the Flags associated to the CSV SplFileObject
     *
     * @param int $flags
     *
     * @throws \InvalidArgumentException If the argument is not a valid integer
     *
     * @return $this
     */
    public function setFlags($flags)
    {
        if (false === filter_var($flags, FILTER_VALIDATE_INT, ['options' => ['min_range' => 0]])) {
            throw new InvalidArgumentException('you should use a `SplFileObject` Constant');
        }

        $this->flags = $flags|SplFileObject::READ_CSV|SplFileObject::DROP_NEW_LINE;

        return $this;
    }
````

Could you confirm this behaviour? To me, this looks like a bug - although I'm not sure if I'm missing something.

Cheers
Pascal

**Comment by nyamsprod** [comment]
What you are referring to is a [known SplFileObject bug reported here](https://bugs.php.net/bug.php?id=69181). The reason the package uses the `DROP_NEW_LINE` flag is because otherwise the last empty line would be added when making a simple  `foreach`. 
A workaround could be to enable the removal of the `DROP_NEW_LINE` flag when using the `setFlags` method.
Having said that, either way this should be documented. It depends on how much user friendly we want the package to be.

**Comment by nyamsprod** [comment]
Since I can not directly fix the `SplFileObject` bug. I've updated the code to give the possibility to the user to remove the `DROP_NEW_LINE` flag. I have updated [the documentation](http://csv.thephpleague.com/properties/#the-splfileobject-flags) as well. When I finish code review for possible breakage I'll make a `7.0.1` bug fix release, today or tomorrow. In the meantime the code is already present in the dev-master branch, if you want to review it yourself.
